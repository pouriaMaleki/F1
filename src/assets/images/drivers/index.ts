import driverImageFernando from "./alonso.png";
import driverImageKimi from "./raikkonen.jpg";
import driverImageHamilton from "./hamilton.png";
import driverImageVettel from "./vettel.png";
import driverImageButton from "./button.png";

export default {
  alonso: driverImageFernando,
  raikkonen: driverImageKimi,
  hamilton: driverImageHamilton,
  vettel: driverImageVettel,
  button: driverImageButton
};
