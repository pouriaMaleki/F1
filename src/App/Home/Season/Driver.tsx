import React from "react";
import drivers from "../../../assets/images/drivers";
import css from "./Driver.module.sass";

interface IProps {
  driver: string;
  className?: string;
}

const Driver = (props: IProps) => (
  <div className={props.className}>
    <img src={drivers[props.driver]} className={css.image} alt={props.driver} />
    <div className={css.title}>{props.driver}</div>
  </div>
);

export { Driver };
