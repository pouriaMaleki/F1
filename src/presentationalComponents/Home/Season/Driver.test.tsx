import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { Driver } from "./Driver";
import drivers from "../../../assets/images/drivers";

const driver = Object.keys(drivers)[0];

describe("Render Driver", () => {
  // snapshot test
  it("render correctly Driver component", () => {
    const DriverComponent = renderer
      .create(<Driver driver={driver} driverId={driver} />)
      .toJSON();
    expect(DriverComponent).toMatchSnapshot();
  });

  // test props
  it("add className as props", () => {
    const CLASSNAME = "CLASSNAME";
    const props = {
        driver: driver,
        driverId: driver,
        className: CLASSNAME
      },
      DriverComponent = mount(<Driver {...props} />);

    expect(DriverComponent.hasClass(CLASSNAME)).toEqual(true);
  });

  it("render proper driver image", () => {
    const props = {
        driver: driver,
        driverId: driver
      },
      DriverComponent = mount(<Driver {...props} />);
    expect(DriverComponent.find("img").prop("src")).toEqual(drivers[driver]);
  });
});
