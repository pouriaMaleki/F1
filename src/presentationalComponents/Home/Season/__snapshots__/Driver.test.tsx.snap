// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Driver render correctly Driver component 1`] = `
<div>
  <img
    alt="alonso"
    className="image"
    src="alonso.png"
  />
  <div
    className="title"
  >
    alonso
  </div>
</div>
`;
